iothub/service/tests/DigitalTwinsClientTests.cs

```diff
@@ -0,0 +1,471 @@
+        {
+            // arrange
+            string digitalTwinId = "foo";
+            string commandName = "foo";
+            string componentName = "bar";
+            var digitalTwin = new BasicDigitalTwin
+            {
+                Id = digitalTwinId,
+            };
+
+            var mockCredentialProvider = new Mock<IotHubConnectionProperties>();
+            mockCredentialProvider
+                .Setup(getCredential => getCredential.GetAuthorizationHeader())
+                .Returns(s_validMockAuthenticationHeaderValue);
+            var mockHttpRequestFactory = new HttpRequestMessageFactory(s_httpUri, "");
+
+            using var mockHttpResponse = new HttpResponseMessage
+            {
+                StatusCode = HttpStatusCode.OK,
+                Content = HttpMessageHelper.SerializePayload(digitalTwin),
+            };
+            mockHttpResponse.Headers.Add("x-ms-command-statuscode", "200");
+            mockHttpResponse.Headers.Add("x-ms-request-id", "200");
+
+            var mockHttpClient = new Mock<HttpClient>();
+            mockHttpClient
+                .Setup(restOp => restOp.SendAsync(It.IsAny<HttpRequestMessage>(), It.IsAny<CancellationToken>()))
+                .ReturnsAsync(mockHttpResponse);
+
+            var digitalTwinsClient = new DigitalTwinsClient(
+                HostName,
+                mockCredentialProvider.Object,
+                mockHttpClient.Object,
+                mockHttpRequestFactory,
+                s_retryHandler);
+
+            // act
+            InvokeDigitalTwinCommandResponse response = await digitalTwinsClient.InvokeComponentCommandAsync(digitalTwinId, componentName, commandName);
+
+            // assert
+            response.Status.Should().Be(200);
+        }
+
+        [TestMethod]
+        [DataRow(null, "foo", null)]
+        [DataRow("bar", null, null)]
+        [DataRow(null, null, "baz")]
+        [DataRow(null, null, null)]
+        public async Task DigitalTwinsClient_InvokeComponentCommandAsync_NullParamater_Throws(string digitalTwinId, string commandName, string componentName)
+        {
+            // arrange
+            using var serviceClient = new IotHubServiceClient(s_connectionString);
+            DigitalTwinsClient digialTwinsClient = serviceClient.DigitalTwins;
+
+            // act
+            Func<Task> act = async () => await digialTwinsClient.InvokeComponentCommandAsync(digitalTwinId, componentName, commandName);
+
+            await act.Should().ThrowAsync<ArgumentNullException>();
+        }
+
+        [TestMethod]
+        public async Task DigitalTwinsClient_InvokeComponentCommandAsync_HttpException()
+        {
+            // arrange
+            string commandName = "test";
+            string componentName = "test";
+            string digitalTwinId = "foo";
+            var digitalTwin = new BasicDigitalTwin
+            {
+                Id = digitalTwinId,
+            };
+            var responseMessage = new ResponseMessage2
```

Comment:
We still have a type called `ResponseMessage2`?

@timtay-microsoft wasn't this a temporary name while we were refactoring?

Reply:
I had no involvement in this class. Git history shows You, @abhipsaMisra and @ngastelum-ms as the ones who have touched this class. I agree that the type name is terrible though, yeah.